File: Python/publisher.py
import paho.mqtt.client as mqtt
import time

# Configurações do Broker
broker = "test.mosquitto.org"
porta = 1883
topico = "siChat/mqtt"

# Função para quando a conexão for estabelecidade com broker
def on_connect(client, userdata, flag, rc):
    print(f"Conectado ao broker com código de resultado {rc}")

# Criar client MQTT
client = mqtt.Client()

# Definir callback de conexão
client.on_connect = on_connect

# Conectar o broker
client.connect(broker, porta, 60)

# Iniciar o loop da rede para processar mensagens
client.loop_start()

print("Precione Ctrl + C para sair: ")
try:
    while True:
        mensagem = input("Digite a mensagem que deseja enviar: ")

        # Publicar a mensagem no tópico
        client.publish(topico, mensagem)
        print (f"Mensagem enviada: {mensagem}")
        time.sleep(1) # Estipula um tempo para suspender e evitar sobrecarga!
except KeyboardInterrupt:
    print("Saindo...")

# Encerrar o loop de rede e desconectar
client.loop_stop()
client.disconnect()

File: JavaScript/publisher.js
const mqtt = require('mqtt')

// Configuração do broker
const broker = "mqtt://test.mosquitto.org"
const topico = 'siChat/mqtt'

// Conectar ao broker
const client = mqtt.connect(broker)

client.on('connect', () => {
    console.log('Conectado ao broker')

    // Ler mensagens do terminal e publicar no tópico MQTT
    process.stdin.on('data', (data) => {
        const mensagem = data.toString().trim();
        client.publish(topico, mensagem)
        console.log(`Mensagem enviada: ${mensagem}`)
    })
})

client.on('error', (err) => {
    console.log('Erro de conexão: ', err)
    client.end()
})